util/src/main/java/io/kubernetes/client/util/ClientBuilder.java

```diff
@@ -100,28 +99,46 @@ public static ClientBuilder standard() throws IOException {
     return standard(true);
   }
 
+  public static ClientBuilder context(String context) throws IOException {
+    final File kubeConfig = getKubeConfigFile();
+    ClientBuilder clientBuilder = getClientBuilder(true, kubeConfig, context);
+    if (clientBuilder != null) return clientBuilder;
+    final File clusterCa = new File(SERVICEACCOUNT_CA_PATH);
+    if (clusterCa.exists()) {
+      return cluster();
+    }
+    return new ClientBuilder();
+  }
+
   public static ClientBuilder standard(boolean persistConfig) throws IOException {
-    final File kubeConfig = findConfigFromEnv();
-    ClientBuilder clientBuilderEnv = getClientBuilder(persistConfig, kubeConfig);
-    if (clientBuilderEnv != null) return clientBuilderEnv;
-    final File config = findConfigInHomeDir();
-    ClientBuilder clientBuilderHomeDir = getClientBuilder(persistConfig, config);
-    if (clientBuilderHomeDir != null) return clientBuilderHomeDir;
+    final File kubeConfig = getKubeConfigFile();
+    ClientBuilder clientBuilder = getClientBuilder(persistConfig, kubeConfig, null);
+    if (clientBuilder != null) return clientBuilder;
     final File clusterCa = new File(SERVICEACCOUNT_CA_PATH);
     if (clusterCa.exists()) {
       return cluster();
     }
     return new ClientBuilder();
   }
 
-  private static ClientBuilder getClientBuilder(boolean persistConfig, File kubeConfig)
+  private static File getKubeConfigFile() {
```

Comment:
This is messy. We should really refactor the whole `ClientBuilder` class so that it defers all of this construction until the `build()` method is called.

That's beyond the scope of this PR though (unless you want to :)) but please add a `TODO: defer construction to build() method` here.